src/components/PaletteContainer/PaletteContainer.js

```diff
@@ -88,7 +88,11 @@ export class PaletteContainer extends Component {
   }
 
   generateRandomHex = () => {
-    return '#' + Math.floor(Math.random() * 16777215).toString(16);
+    let hexCode = '#' + Math.floor(Math.random() * 16777215).toString(16);
+      if(hexCode.length < 7 ) {
+        return this.generateRandomHex()
+      }
+    return hexCode;
```

Comment:
A bit of simple recursion here. Checking the length of the string to make sure it is 7 characters long. Otherwise leading 0s were disregarded and the function was returning incomplete hex codes.